src/System.IO.FileSystem/src/System/IO/Enumeration/FileSystemEntry.Unix.cs

```diff
@@ -112,11 +117,17 @@ public ReadOnlySpan<char> FileName
         /// </summary>
         public ReadOnlySpan<char> OriginalRootDirectory { get; private set; }
 
-        public FileAttributes Attributes => _status.GetAttributes(FullPath, FileName);
-        public long Length => _status.GetLength(FullPath);
-        public DateTimeOffset CreationTimeUtc => _status.GetCreationTime(FullPath);
-        public DateTimeOffset LastAccessTimeUtc => _status.GetLastAccessTime(FullPath);
-        public DateTimeOffset LastWriteTimeUtc => _status.GetLastWriteTime(FullPath);
+        // Windows never fails getting attributes, length, or time as that information comes back
+        // with the native enumeration struct. As such we must not throw here.
```

Comment:
Shouldn't we then get the information at the same time, rather than querying lazily when accessing the property? Or is the difference not visible to the dev?

Reply:
> There is a very small window of opportunity for it to change

I don't feel strongly about it, but I just want to clarify the "very small" part of this.  If I do:
```C#
FileSystemInfo[] entries = Directory.GetFileSystemEntries(...);
Thread.Sleep(TimeSpan.FromHours(3).TotalMilliseconds);
Console.WriteLine(entries[0].Attributes);
```
is Attributes computed at the beginning of this operation or three hours later when I access the Attributes?

Reply:
It is computed 3 hours later, but, as I stated, I don't expect that sort of usage to be typical. Additionally I've mitigated this with Attributes by only changing the ReadOnly flag lazily. That should make logic errors pretty remote.  Timestamps, length, and the ReadOnly flag can change depending on when you first access the data- but they will never change after accessing the first time.

I could add another option to force the attempt up front. It still has no guarantees, of course. What do you think? I can tag the main issue with such a suggestion for API review discussion...

Reply:
> What do you think?

I don't think it's worth new API, especially since that would be odd on the Windows side where we'd then need to recompute it later.

> I don't expect that sort of usage to be typical

Ok.  I just want to make sure we're consciously accepting the difference in behavior.

> There is overhead in getting this info

I assume you've measured it and it's been non-trivial? I can imagine that'd be the case, given that we'd be making another syscall per returned object, but we should verify; if it happens to turn out that it's not a huge overhead, we could simply make the call and avoid any difference.

Reply:
> especially since that would be odd on the Windows side where we'd then need to recompute it later

Not sure I follow. I was thinking of enforcing the stat call right after we get the entry from ReadDir.

> I assume you've measured it and it's been non-trivial? 

Yes, just validated again. It slows enumeration by a factor of 3. :/

Reply:
> It slows enumeration by a factor of 3

Thanks for confirming.

> I was thinking of enforcing the stat call right after we get the entry from ReadDir.

My point was that there are two behaviors: get the value while enumerating, or lazily get the value on first use (or every use). I would expect an API here (which, to be clear, I don't think is warranted) would let you pick between the two or three. And if you selected lazy, the Windows implementation would need to adhere to that. Seems like unnecessary complication.

Reply:
> My point was that there are two behaviors

I see. I didn't think it was necessary to try and match.

Well, in any case, it is pretty easy to get tighter population of the properties- just need to access them when you get the FileSystemEntry. You can always refresh the data later by getting the Info object out. It does create an extra allocation, but only one.